numpyro/mcmc.py

```diff
@@ -701,7 +701,7 @@ def _single_chain_mcmc(self, init, collect_fields=('z',), collect_warmup=False,
         if len(collect_fields) == 1:
             samples = (samples,)
         samples = dict(zip(collect_fields, samples))
-        samples['z'] = self.constrain_fn(samples['z'])
+        samples['z'] = vmap(self.constrain_fn)(samples['z'])
```

Comment:
Without vmap, we rely on broadcasting behaviour of transforms. In some examples, broadcasting does not work.